module/private_quarters/private_quarters.py

```diff
@@ -0,0 +1,289 @@
+import module.config.server as server
+from module.base.timer import Timer
+from module.handler.assets import POPUP_CANCEL
+from module.logger import logger
+from module.ocr.ocr import Digit, DigitCounter
+from module.private_quarters.assets import *
+from module.ui.page import page_private_quarters
+from module.ui.ui import UI
+
+OCR_DAILY_COUNT = DigitCounter(PRIVATE_QUARTERS_DAILY_COUNT, letter=(255, 247, 247), threshold=64)
+if server.server != 'jp':
+    OCR_SHOP_GOLD_COINS = Digit(PRIVATE_QUARTERS_SHOP_GOLD_COINS, letter=(239, 239, 239), name='OCR_SHOP_GOLD_COINS')
+else:
+    OCR_SHOP_GOLD_COINS = Digit(PRIVATE_QUARTERS_SHOP_GOLD_COINS, letter=(201, 201, 201), name='OCR_SHOP_GOLD_COINS')
+
+class PrivateQuarters(UI):
+    # Key: str, target ship name
+    # Value: Button, button instance
+    available_targets = {
+        'anchorage': PRIVATE_QUARTERS_SHIP_ANCHORAGE,
+        'noshiro': PRIVATE_QUARTERS_SHIP_NOSHIRO,
+        'sirius': PRIVATE_QUARTERS_SHIP_SIRIUS,
+    }
+
+    def _pq_target_appear(self):
+        """
+        Callable wrapper to validate target's appearance
+        offset=(100, 100) detectable for anchorage, noshiro, and sirus
+        When more ships added may need to adjust or capture specific bubble position per
+        ship, can use the available_targets to store similarly into tuples instead
+        """
+        settle_timer = Timer(1.5, count=3).start()
+        settle_timer.wait()
+        self.device.screenshot()
+        return self.appear(PRIVATE_QUARTERS_ROOM_TARGET_CHECK, offset=(100, 100))
```

Comment:
wait with image detection, don't use plain wait

Reply:
do you mean use a regular while loop with screen shot arrangement or is there a module function that can help with this?

the reason for the wait is because the ship girl is not guaranteed to appear on room enter, sirius still does this, and anchorage jumps around so will mistake for the ship girl not being there

didn't use a while loop since needed reasons for it to return true or false

otherwise using the room as static elements but would require capturing image for each ship girl's room, seemed overkill in that sense